### DISCUSSION TITLE
VAE - Artifacts Problem

### DISCUSSION THREAD

**Discussion by Madxis** [lllyasviel/Fooocus#2145] [Q&A] [answered]
I have Checkpoints that generate artifacts and they recommend using VAEs to avoid it, but I don't know how to choose a specific one. In the VAE folder I have 2, and although they are in the corresponding folder, I still see artifacts.

In total I have come across 3 checkpoints with this artifact problem.

![0f85bfyy0mfc1](https://github.com/lllyasviel/Fooocus/assets/109711818/1ec00677-2d14-41a2-b23f-85213050840a)

The checkpoint of the preview I uploaded is this: https://civitai.com/models/120090?modelVersionId=268240
_(Euler Karras, CFG 7, 20 steps, seed 5429448612200702317) 
And as you can see in the comments, everyone has solved the problem with VAE.
This is another checkpoint with artifacts that I wasn't able to make it work:
https://civitai.com/models/142313/aikimixl?modelVersionId=157794

So I was wondering how to solve this, since I'm finding more and more checkpoints that need VAEs to function properly.

**Comment by poor7** [comment]
Is another Sampler not suitable for you or gives the same result?

**Reply by Madxis** [reply]
They have a very particular style that is difficult to emulate with another checkpoint (At least I haven't managed it), even so there should be some way to make the VAE work to be able to use any checkpoint, right?

**Comment by 3Diva** [comment]
I haven't tried this model, so I'm not sure if this would help - but as poor7 pointed out, perhaps the sampler will make a difference. I see that in most of the images that the model's creator posted it looks like they're using "Euler a" as the sampler, while the default sampler for Fooocus is "dpmpp_2m_sde_gpu". You might try using the settings the model's creator is using and see if that helps. They're using Euler a which is called "euler_ancestral" in Fooocus, as well as a sampling step count of 20. 

To change the Sampler in Fooocus you can do that under the "Advanced" tab >> "Developer Debug Mode" >> "Sampler". You can change it to "euler_ancestral" to see if that helps. You can also change the step count under Developer Debug Mode and "Forced Overwrite of Sampling Step".

Hopefully that will help with the issue. :)

**Reply by Madxis** [reply]
Hello 3Diva! Thanks for your answer!

That's exactly what I did, in my post you will see the sampler I use, I thought it was that, but I was getting artifacts with both, Euler A and Euler. Here again is the configuration I used for the imagen above: Euler Karras, CFG 7, 20 steps, seed 5429448612200702317
I also tried with all the Scheduler available on Fooocus.

Also, this is what I was answer from someone who fixed the problem:

> First of all you need to download a VAE model and put it in models/VAE in stable diffusion. I use this one:
> https://huggingface.co/stabilityai/sd-vae-ft-ema-original/resolve/main/vae-ft-ema-560000-ema-pruned.ckpt
> 
> Second, you need to change settings in webui under VAE to:
> 
> select the VAE model.
> TAESD
> TAESD
> 
> CLICK APPLY.
> 
> Using these settings you can use ADetailer and generate stuff.
> 
> I use these parameters to generate:
> DPM++ SDE Karras, 4 to 10 steps max
> CFG scale 1.5-3.0
> ADetailer on
> 
> Works pretty good. Fingers and feet are not the best always.

**Reply by 3Diva** [reply]
@Madxis I apologize for not understanding your post - I guess I didn't read it carefully enough. As for changing the VAE that Fooocus uses, I think that's a bit more "under the hood" work than I currently have skills for. There's probably a way to get Fooocus to load a different VAE by modifying the config file but that can be dangerous if one doesn't know what they are doing (and I do not, unfortunately lol). I'm sorry, I wish I could be more help.

**Reply by Madxis** [reply]
Don't worry, I really appreciate you trying to help me :)

Since I am very stubborn, I downloaded the checkpoint again and took one of the images of the creator and tried it again with all the schedulers, same configuration, same prompts and same seed as this one:
https://civitai.com/images/4723478
I also downloaded the same **embedding** just in case.

All the same result, from **normal** to **LCM** scheduler (turbo wasn't working), using all the same as the autor.
I removed the default effects from Fooocus (when you open it you have FooocusV2, Enhance and Sharp selected)

![Karras](https://github.com/lllyasviel/Fooocus/assets/109711818/95432dfa-b2d7-4f01-ba0c-2597714c0812)

I notice that I'm not even getting the same graphic effect as the author, since he/she achieved a more toon/3D style.

And also, if someone is curious, this is the result from the settings the person who fixed the problem gave me:
(dpm++ sde Karras, 10 Steps, CFG scale 2)

![1](https://github.com/lllyasviel/Fooocus/assets/109711818/b1ac4dc1-0b42-46e2-8461-30aef76d0d11)

Again, not even the same style as the original one from the author.
So if someone has any clue to solve the mystery, I would love to try!
But I'm almost sure is something with VAE

**Reply by 3Diva** [reply]
Here's the image I got:
![2024-02-03_23-07-56_8068](https://github.com/lllyasviel/Fooocus/assets/51475772/df4bb80f-f136-41cf-be2e-c523ef8bd869)

Here are the settings I used:
![Capture01](https://github.com/lllyasviel/Fooocus/assets/51475772/aea6a3fc-35b6-4dd3-879a-44d334948e1b)

![Capture02](https://github.com/lllyasviel/Fooocus/assets/51475772/b7d4e9a7-b1e2-4bb7-8ed3-b48874a4d7d3)


The style does look a bit different.

**Reply by 3Diva** [reply]
These settings look a bit better:
![2024-02-03_23-26-32_1268](https://github.com/lllyasviel/Fooocus/assets/51475772/7b785df3-de66-4380-b1e1-92c6dac8b01b)
![Capture04](https://github.com/lllyasviel/Fooocus/assets/51475772/bb9d8a33-d6a2-4350-8909-af2f4c30f5d9)

**Reply by Madxis** [reply]
Thank you very much for taking the time to do this!
I see that you also have artifacts in them but in much smaller quantities
And the colors you get are much more vivid, did you leave Fooocus V2, Enhance and Sharp selected?

**Reply by 3Diva** [reply, marked as answer]
@Madxis No, I unchecked all the styles as I didn't want it to effect the outcome of the image. There are definitely still artifacts but it's not as bad. I do think there's a way to change the VAE but hopefully there are also internal settings that can be found to fix the issue. It seams that the Model is perhaps a little "over trained"? I'm not sure, but I've never seen that issue within Fooocus before, so I'm sure it must have something to do with the model or how it was trained. Hopefully you'll find a better solution or a model that works a little bit easier with Fooocus. :)

**Reply by Madxis** [reply]
Thank you so much! I'm going to save the checkpoint page, hopefully the creator makes a version that works better with Fooocus or some Fooocus update allows us to use VAEs more easily, thank you again!

**Reply by mald55** [reply]
I have the exact same problem with a model I use, and I believe it is the fact it doesn't have an embedded VAE. Not sure on how to fix on Fooocus.

![2024-02-07_12-33-17_7872](https://github.com/lllyasviel/Fooocus/assets/45581991/720a8f06-0e2c-4c11-bbc6-4a77371c380f)

**Reply by Madxis** [reply]
These are exactly the dots I have when trying to use this model: https://civitai.com/models/142313
It's very frustrating because I see others being able to use these models without any artifacts.
As I told 3Diva, I hope that in the future we will be able to switch between different types of VAEs, as I see that being the solution in all cases.
